scripts/commitcheck.sh

```diff
@@ -119,11 +84,11 @@ function coverity_fix_commit()
 
     # test each summary line for the proper format
     OLDIFS=$IFS
-    IFS=$'\n'
+    IFS=$(printf '\n')
```

Comment:
This won't work with sh (or even bash I think) as trailing newlines are lost when assigning the result of a subshell to a variable.

If `IFS=$'\n'` is not desired then perhaps the following would do:

```
IFS=$(printf '\nX')
IFS=${IFS%X}
```

See the following examples to see the problem:
```
$ a=$(printf '\n'); echo "[${a}]"
[]
$ a=$(printf '\nX'); a=${a%X}; echo "[${a}]"
[
]
```

Reply:
Erg yeah duh, good catch. checkbashisms complained about `$'\n'` and suggested the printf instead, but clearly that does not actually behave as intended. I can put a literal linefeed instead.

Reply:
Cheers! :)